qa/rpc-tests/test_framework/test_framework.py

```diff
@@ -50,35 +51,27 @@ def setup_chain(self):
         initialize_chain(self.options.tmpdir)
 
     def setup_nodes(self):
-        return start_nodes(4, self.options.tmpdir)
+        self.nodes = start_nodes(4, self.options.tmpdir)
 
-    def setup_network(self, split = False):
-        self.nodes = self.setup_nodes()
-
-        # Connect the nodes as a "chain".  This allows us
-        # to split the network between nodes 1 and 2 to get
-        # two halves that can work on competing chains.
+    def setup_network(self):
+        # setup_network by default provides linear topology ep. 1 - 2 - 3 - ... - N
+        self.setup_nodes()
+        self.numb_of_nodes = len(self.nodes)
 
-        # If we joined network halves, connect the nodes from the joint
-        # on outward.  This ensures that chains are properly reorganised.
-        if not split:
-            connect_nodes_bi(self.nodes, 1, 2)
-            sync_blocks(self.nodes[1:3])
-            sync_mempools(self.nodes[1:3])
+        for idx, _ in enumerate(self.nodes):
+            if idx < (self.numb_of_nodes-1):
+                connect_nodes_bi(self.nodes, idx, idx+1)
 
-        connect_nodes_bi(self.nodes, 0, 1)
-        connect_nodes_bi(self.nodes, 2, 3)
-        self.is_network_split = split
-        self.sync_all()
+        self.is_network_split = False
+        self.sync_all()       
 
-    def split_network(self):
-        """
-        Split the network of four nodes into nodes 0/1 and 2/3.
-        """
+    def split_network(self, id = 1):
+        # Split the network of between adjanced nodes in linear topology ep. nodes 0-1 and 2-3.
```

Comment:
Possibly an assertion on `id` value can be added (e.g. for handling cases of out-of-range).

Reply:
If any method is called with out of range index, the "Unexpected exception caught during testing: list index out of range" is thrown and handled in test_framework.py.

```Split nodes (1)----x   x---(2)
Unexpected exception caught during testing: list index out of range
  File "/home/titusen/workspace/zen/qa/rpc-tests/test_framework/test_framework.py", line 132, in main
    self.run_test()
  File "/home/titusen/workspace/zen/qa/rpc-tests/headers_06.py", line 130, in run_test
    self.split_network(10)
  File "/home/titusen/workspace/zen/qa/rpc-tests/test_framework/test_framework.py", line 71, in split_network
    disconnect_nodes(self.nodes[id], id + 1)
Stopping nodes
Cleaning up
Failed
!!! FAIL: headers_06.py !!!```